src/WaveTrack.cpp

```diff
@@ -981,9 +1035,73 @@ void WaveTrack::ClearAndPaste(double t0, // Start of time to clear
       // Restore cut/split lines
       if (preserve) {
 
-         // Restore the split lines, transforming the position appropriately
-         for (const auto split: splits) {
-            SplitAt(warper->Warp(split));
+         auto attachLeft = [](WaveClip* target, WaveClip* src) 
+         {
+            wxASSERT(target->GetTrimLeft() == 0);
+            if (target->GetTrimLeft() != 0)
+               return;
+
+            auto trim = src->GetPlayEndTime() - src->GetPlayStartTime();
+            target->Paste(target->GetPlayStartTime(), src);
+            target->SetTrimLeft(trim);
+            //Play start time needs to be ajusted after 
+            //prepending data to the sequence
+            target->Offset(-trim);
+         };
+
+         auto attachRight = [](WaveClip* target, WaveClip* src)
+         {
+            wxASSERT(target->GetTrimRight() == 0);
+            if (target->GetTrimRight() != 0)
+               return;
+            
+            auto trim = src->GetPlayEndTime() - src->GetPlayStartTime();
+            target->Paste(target->GetPlayEndTime(), src);
+            target->SetTrimRight(trim);
+         };
+
+         // Restore the split lines and trims, transforming the position appropriately
+         for (const auto& split: splits) {
+            auto at = LongSamplesToTime(TimeToLongSamples(warper->Warp(split.time)));
+            for (const auto& clip : GetClips())
+            {
+               if (clip->WithinPlayRegion(at))//strictly inside
+               {
+                  auto newClip = std::make_unique<WaveClip>(*clip, mpFactory, true);
+
+                  clip->ClearRight(at);
+                  newClip->ClearLeft(at);
```

Comment:
I think the two lines above should be conditional.  Clear old trim data, only if there is something to attach.  Otherwise, you make this behave unlike WaveClip::Split which preserves all sample data in each new clip at the split.

Am I wrong?

Reply:
I don't think you are wrong, because this question is more about desired behavior. WaveTrack::Split creates a new clip copy with other trimmings and offset values, being used within generators/automated actions they can lead to an enormous amount of duplicates, which is not maintainable (right now). I think it's reasonable to preserve original tracks data hidden behind those splits, but not to introduce new hidden data.